client: update async framework

wusyong opened the pull request:
Update reqwest and tokio runtime. This change also fixes the borrow
check errors in client APIs.

th0br0 commented on curl/src/curl.rs:
```diff
@@ -1,14 +1,69 @@
-use trytes::Trit;
+use constants::*;
+//use trytes::Trit;
```
Remove comment.

th0br0 commented on curl/src/curl.rs:
```diff
@@ -1,14 +1,69 @@
+
+pub trait Sponge {
     fn transform(&mut self);
-    /// Resets the sponge's internal state.
```
Please add the comments back in :'(

th0br0 commented on curl/src/curl.rs:
```diff
@@ -1,14 +1,69 @@
+    where T: Clone + Copy + Sized,
+          Curl<T>: Sponge
+{
+    fn absorb(&mut self, trits: &[T], offset_in: usize, length_in: usize) {
```
Where have all my comments gone

th0br0 replied:
Also, I'm not happy with specifying `offset_in` and `length_in`.
That's what slices are for, they already carry the length around.

paulhandy replied:
I suppose that's fair enough.

paulhandy replied:
Also, sorry about your comments. I'll put them back

th0br0 commented on curl/src/lib.rs:
```diff
@@ -1,8 +1,12 @@
+mod indices;
 pub mod constants;
 pub mod curl;
+//pub mod scurl;
```
remove comment

th0br0 commented on curl/src/pair.rs:
```diff
@@ -0,0 +1,58 @@
+use trytes::*;
+use curl::*;
+use indices::*;
+//use std::mem;
```
remove

th0br0 commented on curl/src/pair.rs:
```diff
@@ -0,0 +1,58 @@
+        let mut scratchpad: Vec<BCTrit> = self.state.iter().map(|&c| (c.0, c.1)).collect();
+
+        for _ in 0..NUMBER_OF_ROUNDS {
+            scratchpad = STATE_INDICES
```
`scratchpad = (0..STATE_LENGTH).into_par_iter()`

paulhandy replied:
that does not compile

th0br0 commented on curl/src/simple.rs:
```diff
@@ -2,41 +2,19 @@ use constants::*;
 
         for _ in 0..NUMBER_OF_ROUNDS {
-            let state_space: Vec<Space> = self.state.iter().map(|&c| c as Space).collect();
+            //let state_space: Vec<Space> = self.state.iter().map(|&c| c as Space).collect();
```
remove

th0br0 commented on curl/src/simple.rs:
```diff
@@ -2,41 +2,19 @@ use constants::*;
-            let state_space: Vec<Space> = self.state.iter().map(|&c| c as Space).collect();
+            //let state_space: Vec<Space> = self.state.iter().map(|&c| c as Space).collect();
+            let state_space: Vec<Trit> = self.state.iter().cloned().collect();
+            //scratchpad.clone_from_slice(state_space.as_slice());
```
remove

th0br0 commented on curl/src/tests.rs:
```diff
@@ -51,15 +53,15 @@ pub mod testsuite {
                               99999999999999999999999OLOB99999999999999999999999"
-                .chars()
-                .collect();
+                    .chars()
```
Did you run `rustfmt`?

paulhandy replied:
vim-racer

th0br0 commented on curl/src/tests.rs:
```diff
@@ -51,15 +53,15 @@ pub mod testsuite {
+                    .collect();
 
-            curl.absorb(&trans.trits());
+            curl.absorb(&trans.trits(), 0, trans.trits().len());
```
Referring to the offset/length thing, you'd just do
```rust
let trits = trans.trits();
curl.absorb(&trits); 
```

here

Commit:
client: update async framework

Update reqwest and tokio runtime. This change also fixes the borrow
check errors in client APIs.
